datafusion/expr/src/type_coercion/binary.rs

```diff
@@ -623,6 +640,17 @@ fn null_coercion(lhs_type: &DataType, rhs_type: &DataType) -> Option<DataType> {
     }
 }
 
+fn check_time_unit(datatype: DataType) -> Option<TimeUnit> {
+    use arrow::datatypes::DataType::*;
```

Comment:
**nitpick:** If you're going to use this in every function of this file, you might as well put it at the top of the file

Reply:
I see! I put it above the `temporal_coercion` function where it is used, I guess it is a good compromise. At the very top it would be a bit out of context in my opinion.

Reply:
Well, since we are adding support to `Time32(TimeUnit`), and `TimeUnit` can take 4 different values, we have to make sure that all of them are covered. In case the `TimeUnit` is `Microsecond` or `Nanosecond`, it is not valid for type `Time32` and there is should be no type coercion, so `temporal_coercion` should return None. However, if the `TimeUnit` is `Second` or `Millisecond`, type coercion must happen and it should return `Some(..)`. Analogously for `Time64`. 

To sum up, the distinction between both cases requires a function that checks if the `TimeUnit` is consistent with the `Time` type.